File: jsimple-json/src/main/java/jsimple/json/readerwriter/JsonStringProperty.java
package jsimple.json.readerwriter;

/**
 * @author Bret Johnson
 * @since 5/22/13 9:07 PM
 */
public class JsonStringProperty extends JsonProperty {
    public JsonStringProperty(String name, int id) {
        super(name, id);
    }

    public String readValue(JsonObjectReader objectReader) {
        return (String) objectReader.readPropertyValue();
    }

    @Override public Object readValueUntyped(JsonObjectReader objectReader) {
        return objectReader.readPropertyValue();
    }

    public void write(JsonObjectWriter objectWriter, String value) {
        objectWriter.writeProperty(this, value);
    }
}

File: jsimple-json/c#/jsimple/json/readerwriter/JsonStringProperty.cs
namespace jsimple.json.readerwriter {

    /// <summary>
    /// @author Bret Johnson
    /// @since 5/22/13 9:07 PM
    /// </summary>
    public class JsonStringProperty : JsonProperty {
        public JsonStringProperty(string name, int id) : base(name, id) {
        }

        public virtual string readValue(JsonObjectReader objectReader) {
            return (string) objectReader.readPropertyValue();
        }

        public override object readValueUntyped(JsonObjectReader objectReader) {
            return objectReader.readPropertyValue();
        }

        public virtual void write(JsonObjectWriter objectWriter, string value) {
            objectWriter.writeProperty(this, value);
        }
    }

}